src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -89,49 +63,81 @@ def generate_requests(
                 )
         return request_params
 
+    @staticmethod
+    def assign_placeholders(value: str, param_values: Dict[str, Any]) -> str:
+        """
+        Finds all the placeholders (indicated by <>) in the passed in value
+        and replaces them with the actual param values
+        """
+        if value:
+            placeholders = re.findall("<(.+?)>", value)
+            for placeholder in placeholders:
+                value = value.replace(
+                    f"<{placeholder}>", str(param_values[placeholder])
+                )
+        return value
+
+    def map_param_values(
+        self, current_request: SaaSRequest, param_values: Dict[str, Any]
+    ) -> SaaSRequestParams:
+        """
+        Visits path, headers, and query_params in the current request and replaces
+        the placeholders with the request param values
+        """
+
+        path: str = self.assign_placeholders(current_request.path, param_values)
+
+        headers: Dict[str, Any] = {}
+        for header in current_request.headers or []:
+            headers[header.name] = self.assign_placeholders(header.value, param_values)
+
+        query_params: Dict[str, Any] = {}
+        for query_param in current_request.query_params or []:
+            query_params[query_param.name] = self.assign_placeholders(
+                query_param.value, param_values
+            )
+
+        return SaaSRequestParams(
+            method=current_request.method,
+            path=path,
+            headers=headers,
+            query_params=query_params,
+        )
```

Comment:
I like how you're consolidating a couple locations where SaasRequestParams are being constructed, when we're editing, it's easy to forget how many places these are being built (with the pagination ones too)